dig.go

```diff
@@ -73,26 +72,28 @@ type Container struct {
 	// index map[reflect.Type]key
 }
 
-// New constructs a ready-to-use Container.
+// New constructs a Container.
 func New(opts ...Option) *Container {
 	return &Container{
 		nodes: make(map[key]*node),
 		cache: make(map[key]reflect.Value),
 	}
 }
 
-// Provide teaches the Container how to construct one or more new types.
+// Provide teaches the container how to build values of one or more types.
 //
-// Any function passed to Provide is assumed to be a constructor. Constructors
-// can take any number of parameters, which will be supplied by the Container
-// on demand. They must return at least one non-error value, all of which are
-// then available in the Container. If the last returned value is an error, the
-// Container inspects it to determine whether the constructor succeeded or
-// failed. Regardless of position, returned errors are never put into the
-// Container's dependency graph.
+// The first argument of Provide is a function which accepts zero or more
+// parameters and returns one or more results. The function may optionally
+// return an error to indicate that it failed to build the value. This
+// function will be treated as the constructor for all the types it returns.
+// This function will be called AT MOST ONCE when a type produced by it, or a
+// type which depends on this function's output, is requested via Invoke. If
+// the same types are requested multiple times, the previously produced value
+// will be reused.
 //
-// All non-functions (including structs, pointers, Go's built-in collections,
-// and primitive types like ints) are inserted into the Container as-is.
+// In addition to accepting vanilla constructors, Provide also accepts
+// constructors which accept dig.In structs as dependencies and/or return
```

Comment:
I would use "take" here instead of "accept", since you already used "accepting" earlier in the sentence.